### PR TITLE
Addon Remark-GFM: Upgrade remark-gfm

### PR DESCRIPTION
**Pull request by yannbf**
Relates to #25182 

## What I did

Upgraded remark-gfm from v3 to [v4](https://github.com/remarkjs/remark-gfm/releases/tag/4.0.0), which contains a few possible breaking changes, but mostly Node 16 being the minimum version

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Review by shilman — APPROVED** [review]

File: code/lib/codemod/src/transforms/mdx-to-csf.ts [hunk 1]
```diff
@@ -136,6 +136,8 @@ export function transform(source: string, baseName: string): [string, string] {
             value: `/* ${nodeString} is deprecated, please migrate it to <Story of={referenceToStory} /> see: https://storybook.js.org/migration-guides/7.0 */`,
           };
           storiesMap.set(idAttribute.value as string, { type: 'id' });
+          // @ts-ignore need to check with @kasperpeulen, this happened with the upgrade of
```
Comment (yannbf) [inline code/lib/codemod/src/transforms/mdx-to-csf.ts:139]: @kasperpeulen I need assistance here <3

**Review by valentinpalkovic — APPROVED** [review]

**Comment by ndelangen** [comment]
Are we keeping this addon around for 8.0 actually?

**Comment by yannbf** [comment]
> Are we keeping this addon around for 8.0 actually?

Yes, I spoke to @shilman and the decision was to have this stick a little longer so users can have a smoother migration process. If you feel like we should reconsider, please let us know!

**Comment by shilman** [comment]
I want to support a direct 6.5 to 8.0 upgrade since roughly half our users are still on 6.5. We can drop this in 9.0.

**Pull request merged**